poetry/mixology/version_solver.py

```diff
@@ -40,7 +39,7 @@ class VersionSolver:
     def __init__(
         self,
         root,  # type: ProjectPackage
-        provider,  # type: Provider
+        provider,
```

Comment:
Is there a reason why we are dropping the type hint here?

Reply:
I don't remember exactly. I thin I got a cyclic import error at some point due to this import. I can check again to see if we really need to remove it.

Reply:
So, I just checked and we need to remove it to avoid a cyclic import error